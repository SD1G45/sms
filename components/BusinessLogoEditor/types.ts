export interface ProfilePictureEditorProps {
  onClose: () => void;
}
