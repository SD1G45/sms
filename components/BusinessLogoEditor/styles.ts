import styled from "styled-components";
import Button from "../Button";
import Image from "next/image";

export const Container = styled.div`
  background: white;
`;

export const TopSection = styled.div`
  display: flex;
  justify-content: flex-end;
  width: 100%;
  padding: 10px 10px 0 0;
`;

export const BottomSection = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 30px;
`;

export const SaveButton = styled(Button)`
  padding: 5px 10px;
  margin-top: 20px;
`;

export const UploadButton = styled(SaveButton)`
  border: 2px solid ${(props) => props.theme.colors.primary};
  margin-bottom: 10px;
`;

export const ErrorText = styled.p`
  color: red;
  font-size: 1.3rem;
  margin-top: 5px;
`;

export const CloseButon = styled(Image)`
  width: 30px;
  height: 30px;
  justify-content: center;
  cursor: pointer;
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 30px;
  width: 100%;
`;
