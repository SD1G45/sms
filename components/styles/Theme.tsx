import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
