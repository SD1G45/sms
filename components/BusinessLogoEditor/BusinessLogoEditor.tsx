import React from "react";
import AvatarEditor from "react-avatar-editor";
import Button from "../Button";
import Slider from "../Slider";
import { useRouter } from "next/router";
import { Container, BottomSection, ErrorText, ButtonContainer } from "./styles";

export interface Props {
  onClose: () => void;
}

const BusinessLogoEditor: React.FC<Props> = ({ onClose }) => {
  const [scaleValue, setScaleValue] = React.useState(1);
  const [file, setFile] = React.useState<any>(null);
  const [errorMsg, setErrorMsg] = React.useState("");
  const [loading, setLoading] = React.useState(false);
  const editor = React.useRef<any>(null);
  const uploadInput = React.useRef<any>(null);
  const router = useRouter();

  const { business_id } = router.query;

  const handleUpload = (ev: any) => {
    let file = uploadInput.current.files[0];
    setFile(file);
  };

  const onClickSave = async () => {
    if (editor && file !== null) {
      setLoading(true);
      setErrorMsg("");

      const data = {
        base64: editor.current.getImage().toDataURL(),
        businessId: business_id,
      };

      const rootUrl =
        process.env.NODE_ENV === "development"
          ? "http://localhost:3001/"
          : "https://www.smsasdflkjasdf.co/";

      fetch(rootUrl + "api/picture", {
        method: "POST",
        credentials: "include",
        headers: { "content-type": "application/json" },
        body: JSON.stringify(data),
      })
        .then((res) => {
          setLoading(false);
          onClose();
        })
        .catch(() => {
          setLoading(false);
          setErrorMsg(
            "There was an error uploading your profile picture. Please try again later."
          );
        });
    } else {
      setErrorMsg("Please select an image.");
    }
  };

  return (
    <Container>
      <BottomSection>
        <AvatarEditor
          ref={editor}
          image={file}
          width={250}
          height={250}
          border={10}
          borderRadius={10000}
          color={[0, 0, 0, 0.5]} // RGBA
          scale={scaleValue}
          rotate={0}
        />
        <Slider
          value={scaleValue}
          onChange={(event: any) => setScaleValue(event.target.value)}
        />
        <input
          onChange={handleUpload}
          ref={uploadInput}
          type="file"
          style={{ textAlign: "right" }}
        />
        <ButtonContainer>
          <Button invert onClick={onClose}>
            Cancel
          </Button>
          <Button onClick={onClickSave} style={{ marginLeft: 15 }}>
            Save
          </Button>
        </ButtonContainer>
        {errorMsg.length > 0 ? <ErrorText>{errorMsg}</ErrorText> : <div />}
      </BottomSection>
    </Container>
  );
};

export default BusinessLogoEditor;
