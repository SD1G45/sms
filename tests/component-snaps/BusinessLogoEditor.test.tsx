import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import BusinessLogoEditor from "../../components/BusinessLogoEditor";
import Theme from "../../components/styles/Theme";
import "jest-canvas-mock";
import AvatarEditor from "react-avatar-editor";

jest.mock("react-avatar-editor");

jest.mock("next/router", () => ({
  useRouter() {
      return {
          route: "/",
          pathname: "",
          query: "",
          asPath: "",
      };
  },
}));

describe("<BusinessLogoEditor/>", () => {
  it("renders the editor", () => {
    const tree = renderer.create(
      <Theme>
        <BusinessLogoEditor onClose={ () => {} }/>
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});