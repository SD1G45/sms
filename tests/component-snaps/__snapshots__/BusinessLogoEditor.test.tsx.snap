// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessLogoEditor/> renders the editor 1`] = `
.c4 {
  width: 100%;
  padding: 12px 14px;
  background-color: white;
  border-radius: 18px;
  border: 2px solid #4881F0;
  color: #4881F0;
  cursor: pointer;
  font-size: 1rem;
  opacity: 100%;
  overflow: hidden;
  white-space: nowrap;
}

.c4:hover {
  opacity: 70%;
}

.c5 {
  width: 100%;
  padding: 12px 14px;
  background-color: #4881F0;
  border-radius: 18px;
  border: 2px solid white;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  opacity: 100%;
  overflow: hidden;
  white-space: nowrap;
}

.c5:hover {
  opacity: 70%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #888;
  margin-top: 2rem;
  margin-bottom: 2rem;
  width: 100%;
}

.c2 .value {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 2rem;
}

.c2 .slider {
  -webkit-flex: 6;
  -ms-flex: 6;
  flex: 6;
  -webkit-appearance: none;
  width: 100%;
  height: 15px;
  border-radius: 5px;
  background: #efefef;
  outline: none;
}

.c2 .slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 25px;
  height: 25px;
  background: #4881F0;
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: .2s;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
}

.c2 .slider::-moz-range-thumb {
  width: 25px;
  height: 25px;
  background: #4881F0;
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: .2s;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
}

.c0 {
  background: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 30px;
  width: 100%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <input
        aria-labelledby="continuous-slider"
        className="slider"
        max={5}
        min={1}
        onChange={[Function]}
        step={0.001}
        type="range"
        value={1}
      />
    </div>
    <input
      id="file-upload"
      onChange={[Function]}
      style={
        Object {
          "textAlign": "right",
        }
      }
      type="file"
    />
    <div
      className="c3"
    >
      <button
        className="c4"
        id="cancel"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="c5"
        id="save"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": 15,
          }
        }
      >
        Save
      </button>
    </div>
    <div />
  </div>
</div>
`;
